packages/console/src/components/CodeEditor/index.tsx

```diff
@@ -1,21 +1,50 @@
-import React, { ChangeEventHandler } from 'react';
+import MonacoEditor from '@monaco-editor/react';
+import React from 'react';
 
 import * as styles from './index.module.scss';
 
-// Will be implemented in LOG-1708, defined 2 basic props for now.
 type Props = {
+  language: string;
+  isDarkMode?: boolean;
+  height?: string;
+  isReadonly?: boolean;
   value?: string;
   onChange?: (value: string) => void;
 };
 
-const CodeEditor = ({ value, onChange }: Props) => {
-  const handleChange: ChangeEventHandler<HTMLTextAreaElement> = (event) => {
-    onChange?.(event.target.value);
+const CodeEditor = ({
+  value,
+  onChange,
+  language,
+  height = '300px',
+  isReadonly = false,
+  isDarkMode,
+}: Props) => {
+  const handleChange = (changedValue: string | undefined) => {
+    onChange?.(changedValue ?? '');
+  };
+
+  // See https://microsoft.github.io/monaco-editor/api/enums/monaco.editor.EditorOption.html
+  const options = {
```

Comment:
you can use typed options from `monaco-editor`

Reply:
updated